### DISCUSSION TITLE
OTA Firmware downgrade for TS011F_plug_1

### DISCUSSION THREAD

**Discussion by Jordan87** [Koenkk/zigbee2mqtt#13059] [Devices]
Hi
I want to perform FW downgrade on my three different (different manufacturers) power plugs.
They were updated with Z2M to 1.0.13 version but because of this issue https://github.com/Koenkk/zigbee2mqtt/issues/11648 is still visible in my case i want to perform downgrade to earlier version and check with Tuya gateway if any updates are available.

I added lines to /config/zigbee2mqtt/configuration.yaml

> ota:
  zigbee_ota_override_location: my_index.json

Base on old version from here https://github.com/Koenkk/zigbee-OTA/commit/07811acd3139b7be497b36cf7d3808db379c0de0 I created my own /config/zigbee2mqtt/my_index.json file

> [
    {
        "fileVersion": 74,
        "fileSize": 305986,
        "sha512": "3c70afc8d2488c22b7a99ab53cf0250b6aa12b2eb5d703cb819620c07dc10dbdd7fe8e3a2ef63cdaedd9b97d502c71c2c2c199fb05121f977b87a86e8bbe91bb",
        "url": "https://github.com/Koenkk/zigbee-OTA/raw/master/images/Telink/nas_wr01b_plug_OTA_1.0.10.bin",
        "path": "images/Telink/nas_wr01b_plug_OTA_1.0.10.bin"
        "manufacturerCode": 4417,
        "imageType": 54179,
        "force": true
    }
]

I also tried with simple local version and file nas_wr01b_plug_OTA_1.0.10.bin in /config/zigbee2mqtt
 
> [
    {
        "url": "nas_wr01b_plug_OTA_1.0.10.bin",
        "force": true
    }
]
 
But still for some reason in log i see only info about online server with latest version

> Zigbee2MQTT:debug 2022-07-06 11:13:29: Received MQTT message on 'zigbee2mqtt/bridge/request/device/ota_update/check' with data '{"id":"Nous gniazdko Drukarka?","transaction":"yv8ev-1"}'
Zigbee2MQTT:info  2022-07-06 11:13:29: Checking if update available for 'Nous gniazdko Drukarka?'
Zigbee2MQTT:debug 2022-07-06 11:13:29: Check if update available for '0xa4c138eddc3a0694' (TS011F)
Zigbee2MQTT:debug 2022-07-06 11:13:29: Using endpoint '1'
Zigbee2MQTT:debug 2022-07-06 11:13:29: Received Zigbee message from 'Nous gniazdko Drukarka?', type 'commandQueryNextImageRequest', cluster 'genOta', data '{"fieldControl":0,"fileVersion":77,"imageType":54179,"manufacturerCode":4417}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2022-07-06 11:13:29: Got OTA request '{"fieldControl":0,"manufacturerCode":4417,"imageType":54179,"fileVersion":77}'
Zigbee2MQTT:debug 2022-07-06 11:13:29: ZigbeeOTA: downloaded main index
Zigbee2MQTT:debug 2022-07-06 11:13:29: Is new image available for '0xa4c138eddc3a0694', current '{"fieldControl":0,"manufacturerCode":4417,"imageType":54179,"fileVersion":77}', latest meta '{"fileVersion":77,"fileSize":307682,"url":"https://github.com/Koenkk/zigbee-OTA/raw/master/images/Telink/1654157434-oem_zg_tl8258_plug_OTA_1.0.13.bin","sha512":"97ea8413e8ab662f2bebf3b013b3030754cbc2c0744b91b14741fff17a2d8d99e7088aa22a3b6e43b3595ccd65433f96d32516fc057fddcc13a816c921af8c97"}'
Zigbee2MQTT:debug 2022-07-06 11:13:29: Update available for '0xa4c138eddc3a0694': NO
Zigbee2MQTT:info  2022-07-06 11:13:29: No update available for 'Nous gniazdko Drukarka?'
Zigbee2MQTT:info  2022-07-06 11:13:29: MQTT publish: topic 'zigbee2mqtt/Nous gniazdko Drukarka?', payload '{"child_lock":"UNLOCK","current":0,"energy":0.49,"indicator_mode":"off/on","linkquality":91,"power":0,"power_outage_memory":"restore","state":"OFF","update":{"state":"idle"},"update_available":null,"voltage":239}'
Zigbee2MQTT:info  2022-07-06 11:13:29: MQTT publish: topic 'zigbee2mqtt/bridge/response/device/ota_update/check', payload '{"data":{"id":"Nous gniazdko Drukarka?","updateAvailable":false},"status":"ok","transaction":"yv8ev-1"}'

Have I missed something?

**Comment by Jordan87** [comment]
Nobody knows how to fix it?
I have one more new device with old FW and it is also detecting standard update to 1.0.13 even with 

> [
{
"url": "nas_wr01b_plug_OTA_1.0.10.bin",
"force": true
}
]

**Comment by Jordan87** [comment]
There was invalid name
instead "zigbee_ota_override_index_location" i used "zigbee_ota_override_location" but i dont know how it was possible that upgade few days back was working with local file. Maybe command/structure was changed in the meantime?

**Comment by AoSpades** [comment]
I would like to downgrade too. But the guide on https://www.zigbee2mqtt.io/devices/TS011F_plug_1.html does not work. If I follow the directions, create a my_index.json with the information provided there the device does not update.
